src/element/textWysiwyg.tsx

```diff
@@ -434,61 +439,57 @@ export const textWysiwyg = ({
     // it'd get stuck in an infinite loop of blur→onSubmit after we re-focus the
     // wysiwyg on update
     cleanup();
-    const updateElement = Scene.getScene(element)?.getElement(element.id);
+    const updateElement = Scene.getScene(element)?.getElement(
+      element.id,
+    ) as ExcalidrawTextElement;
     if (!updateElement) {
       return;
     }
-    let wrappedText = "";
-    if (isTextElement(updateElement) && updateElement?.containerId) {
-      const container = getContainerElement(updateElement);
-
-      if (container) {
-        wrappedText = wrapText(
-          editable.value,
-          getFontString(updateElement),
-          container.width,
-        );
+    let text = editable.value;
+    const container = getContainerElement(updateElement);
+
+    if (container) {
+      text = updateElement.text;
+      const editorHeight = Number(editable.style.height.slice(0, -2));
+      if (editable.value) {
+        // Don't mutate if text is not updated, this happens when text
+        // properties updated since vertically align is taken care in
+        // redrawTextBoundingBox already
+        if (initialText !== editable.value) {
+          mutateElement(updateElement, {
+            // vertically center align
+            y: container.y + container.height / 2 - editorHeight / 2,
+            height: editorHeight,
+            width: Number(editable.style.width.slice(0, -2)),
+            // preserve padding
+            x: container.x + BOUND_TEXT_PADDING,
+            angle: container.angle,
+          });
```

Comment:
Does this serve any purpose now that we're calling `getAdjustedDimensions()` on every change?

Reply:
Actually its not needed since now the coords are updated in `updateWysiwygStyle`